### DISCUSSION TITLE
Orion Browser iPadOS Support

### DISCUSSION THREAD

**Discussion by wizardtoads** [AprilSylph/XKit-Rewritten#1366] [Suggestion Box]
Orion is a webkit based browser that can take Firefox and Chrome extensions, including on iOS/iPadOS. Currently, most browsers for the iOS and iPadOS cannot use addons, including Firefox itself, so this is pretty cool.

Though you can install Xkit Rewritten, neither the Firefox nor the Chrome version currently works on the iPadOS Orion browser. New Xkit does seem to work, and so do other general add ons like Twiitter Control Panel. I don't have an iPhone, so I don't know if that's different.

I don't know what's possible or what would need to be adjusted, but I would love it if I can use XKit Rewritten on my iPad through this browser!

**Comment by wizardtoads** [comment]
Okay spoke a little too soon about New Xkit working, but it does add the New Xkit menu item into Tumblr, as opposed to doing nothing at all, so it does seem Tumblr's interface can be messed with through addons in this browser.

**Comment by marcustyphoon** [comment]
I attempted to figure this out a few weeks ago, but with absolutely no dev tools in the mobile version of Orion, I've got nothing. If they add dev tools and a way to load the development version of an extension I might revisit this.

Hopefully Apple does the rumored policy change of dropping the requirement that iOS browsers use Safari underneath such that Firefox for iOS simply allows XKit Rewritten to work as currently published, though. That would be much simpler.

**Reply by wizardtoads** [reply]
I see! I think they're still pretty new so hopefully they will put out some tools, 

Thanks for the quick reply and all your hard work!!

**Reply by glmn** [reply]
> ###
> [ ](/marcustyphoon)

Looks like they've added devtools 10 days ago in Testflight version.
https://orionfeedback.org/d/3063-activate-the-iphoneipad-debug-console-or-web-inspector

**Reply by marcustyphoon** [reply]
I've been through a few back-and-forth bug report/fix/next bug report loops with them now; at the rate at which these happen I would randomly guess the amount of time before potential XKit support to be months or years. Seems like they're a small team. I'll keep poking at it occasionally.

**Comment by tacticalIndex** [comment]
Could we get developer tools? I kind of need to grab my cookies on some sites.

**Reply by marcustyphoon** [reply]
Not sure what this has to do with this repository, but for the record, Orion on iOS does support Safari developer tools on a Mac connected with a cable to the iOS device.
